needle.pm

```diff
@@ -41,10 +41,8 @@ sub new {
     }
     else {
         local $/;
-        # TODO preserving old behaviour, why should failing read be allowed?
-        no autodie qw(open);
         open(my $fh, '<', $jsonfile);
-        eval { $json = decode_json(<$fh>) };
+        $json = decode_json(<$fh>);
         close($fh);
         if (!$json || $@) {
```

Comment:
`decode_json` modifies `$@`? Is it documented somewhere?
What I found is that If there is an error it will `croak` which if not catched by `eval` you removed will terminate the script.